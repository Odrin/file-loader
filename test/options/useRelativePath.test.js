import webpack from '../helpers/compiler';
import { loaders } from '../helpers/stats';

describe('useRelativePath', () => {
  test('- this.options.context', async () => {
    const config = {
      loader: {
        test: /(png|jpg|svg)/,
        options: {
          useRelativePath: true,
        },
      },
    };

    const stats = await webpack('./file.js', config);
    const { err, src, map, meta } = loaders(stats, 1);

    expect(err).toMatchSnapshot();
    expect(src).toMatchSnapshot();
    expect(map).toMatchSnapshot();
    expect(meta).toMatchSnapshot();
  });

  test('- options.context', async () => {
    const config = {
      loader: {
        test: /(png|jpg|svg)/,
        options: {
          options: {
            context: './relative/',
            useRelativePath: true,
          },
        },
      },
    };

    const stats = await webpack('./file.js', config);
    const { err, src, map, meta } = loaders(stats, 1);

    expect(err).toMatchSnapshot();
    expect(src).toMatchSnapshot();
    expect(map).toMatchSnapshot();
    expect(meta).toMatchSnapshot();
  });
});
