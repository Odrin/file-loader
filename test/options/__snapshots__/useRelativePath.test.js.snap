// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useRelativePath - options.context 1`] = `Array []`;

exports[`useRelativePath - options.context 2`] = `"module.exports = __webpack_public_path__ + \\"9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";"`;

exports[`useRelativePath - options.context 3`] = `undefined`;

exports[`useRelativePath - options.context 4`] = `Object {}`;

exports[`useRelativePath - this.options.context 1`] = `Array []`;

exports[`useRelativePath - this.options.context 2`] = `"module.exports = __webpack_public_path__ + \\"./9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";"`;

exports[`useRelativePath - this.options.context 3`] = `undefined`;

exports[`useRelativePath - this.options.context 4`] = `Object {}`;
